### DISCUSSION TITLE
Question: How do I redirect to custom url(https://myWebApp.com) after logout ?

### DISCUSSION THREAD

**Discussion by 29swastik** [apache/superset#18352] [Q&A / Help]
I integrated superset with my Web Application (**myWebApp.com**). When user login to my website he/she can enter to Superset by clicking a link and superset will open in **new tab** (Behind the scene my code will login user to superset and lands him/her to superset welcome page without showing superset login page). 

So now there are 2 cases which will logout my user from superset 

**CASES**:

**CASE-1.**  When user press logout button from superset website
**CASE-2.** When user logout from my website i.e, **myWebApp.com**

**In both these cases after logout, user should be navigated to my web app's home page (**myWebApp.com**) from that tab where superset was opened. How can I implement this ?**

One more problem is when superset session expires the user will be logout and if the user tries to use superset functionality like viewing dashboards, creating dashboards, viewing profile etc.., the user will get redirected to **superset login page** since he/she is already logged out from superset as show in the image below (Same thing will happen with my **CASE-2** mentioned above).

<img width="1265" alt="Screenshot 2021-11-26 at 7 31 40 PM" src="https://user-images.githubusercontent.com/65583094/143592297-1997d33e-cff7-4c1c-9f0d-4d3c3289bc6d.png">

**Environment**

- Superset version - 1.3.2
- Superset installed in Python virtual environment

Thanks & Regards,
Swastik

**Comment by berlicon** [comment]
You can subscribe to event flask_login.user_logged_out
and do anything.
See https://flask-login.readthedocs.io/en/latest/#flask_login.user_logged_out
